_posts/2021-09-21-release-1.14.0.md

```diff
@@ -0,0 +1,182 @@
+---
+layout: post 
+title:  "Apache Flink 1.14.0 Release Announcement"
+date: 2021-09-21T08:00:00.000Z 
+categories: news 
+authors:
+- joemoe:
+  name: "Johannes Moser"
+
+excerpt: The Apache Flink community is excited to announce the release of Flink 1.14.0! Around xxx contributors worked on over xxxx issues to TODO.
+---
+
+Just a couple of days ago the Apache Software Foundation announced it’s annual report and Apache
+Flink being in the Top 5 in all relevant categories is just an outcome of the work of the
+community (that has been done yet again) for 1.14.0. The consistency how this project is moving
+forward is remarkable. Once again 200 plus contributors worked on over 1,000 issues.
+
+Apache Flink not only supports batch and stream processing, but has been always following the goal
+of making it a unified experience. With Apache Flink 1.14.0 batch and stream processing moved closer
+together. The first sinks and sources are now providing a unified API (following FLIP-27 and
+FLIP-143). Hybrid source has been introduced. The DataStream batch mode has been pushed to the
+TableAPI. Under the hood checkpoints are allowed even after tasks are finished truly enabling mixed
+or bounded jobs. Existing features have been haromised throughout all available APIs. From
+DataStream to TableAPI and SQL and vice versa. The DataStream batch mode is maturing after its
+initial release in 1.13.0.
+
+Fault tolerance is part of Flink’s nature, still it can never be enough. By using the new option of
+debloating the buffers the checkpoint size can decrease significantly and reduce checkpointing times
+to a minimum.
+
+That’s not all there is a huge list of improvements and new additions through out all components.
+Also we had to say goodbye to some features that have been superseded in recent releases. We hope
+you like the new release and we’d be eager to learn about your experience with it, which yet
+unsolved problems it solves, what new use-cases it unlocks for you.
+
+{% toc %}
+
+# Notable improvements for a Unified Batch and Stream Processing experience
+
+Apache Flink unlocks both batch and stream processing use cases. There is a lot of traction for
+both. Initially both features were rather separated, that’s why the API were not really aligned in
+the first place and also moved into apart over time. With both APIs being stable users started to
+combine them in their solutions. Having batch style workloads to initially process historic data and
+then switching into the streaming mode to deal with live data is something that makes sense. But
+having the two APIs separated not only lead to the mentioned differences but also to big white spots
+on the sparsity matrices and it became quite confusing what worked with which API and what not.
+About a year ago the community started to unify the experience by seeing batch as a special case of
+streaming. The notion of bounded and unbounded streams has been introduced and initially released in
```

Comment:
I think, the notion of bounded/unbounded streams and batch being a special case of streaming has always been there. 

It is just that we've been executing batch programs as if they were streaming programs - not making use of their bounded nature to improve performance. This is what batch execution mode addresses.